code/controllers/master.dm

```diff
@@ -84,6 +84,8 @@ GLOBAL_REAL(Master, /datum/controller/master) = new
 /datum/controller/master/New()
 	if(!config)
 		config = new
+	if(!Logger)
+		Logger = new
```

Comment:
Nope. not the right spot for this. When i did my first refactor New and Initialize on master had hundreds of snowflake lines and I am very protective of people dumping world start lines in New(). config is only allowed because it requires the MC existing to work. Logger must not require the MC to exist, or Master to non-null (or hell, it must not require config to exist or be non-null. log types without a config type must work in those situations)

Please go to line 14 in this same file and see how the master and debugger vars are setup, and do the same with logger.